### PR TITLE
Resize images during transload

### PR DESCRIPTION
**Pull request by ermacaz**
Adds configurable options to specify a maximum height or width of an image and resizes it during the transload process.
Currently only for png/jpegs

### CODE REVIEW

File: src/js/transloader.js [hunk 1]
```diff
@@ -157,14 +166,36 @@ function fetchImage(url, callback) {
...
+						// resizeJpeg(arrayBufferView, dimentions[0], dimentions[1],  filesize, mimetype, callback)
+					} else {
+						callback(filesize, mimetype, arrayBufferView);
+					}
+					break;
+				case "image/png":
+					var dv = new DataView(arrayBufferView.buffer)
+					if ((dv.getInt32(16) > MAX_WIDTH) || (dv.getInt32(20) > MAX_HEIGHT)) {
+						resizeImage(blob, mimetype, callback)
+					}
```
Comment (Seiru) [inline src/js/transloader.js]: Need an else with the callback here, if the png is the correct size already

**Review by Seiru — CHANGES_REQUESTED** [review]

**Comment by Seiru** [comment]
This seems to work for resizing pngs, but jpegs don't seem to be resized.

**Comment by Seiru** [comment]
Here's two I was testing with:

https://upload.wikimedia.org/wikipedia/commons/8/8c/Chess_Large.JPG

https://upload.wikimedia.org/wikipedia/commons/d/d5/Dds40-097_large.jpeg

**Comment by ermacaz** [comment]
Ok.  Seems I just got lucky with the JPEGS I was testing on.  The two you found have different type of JPEG encoding... Im going to read up a bit and see how to grab the resolution from this format, or rethink how I get the initial resolution in general.    I will push an update once I figure something out.

Thanks for the help

**Comment by ermacaz** [comment]
OK I added a new commit that simplifies how to get the original resolution regardless of type.  This should fix the jpeg issues and the codepaths that never call the callback

**Pull request merged**